src/types/index.test-d.ts

```diff
@@ -0,0 +1,95 @@
+import { expectType } from 'tsd';
```

Comment:
How about split this file to match/test different modules separately so we don't create a monster test file in the future? 

Also, looks like we are using `dtslint` in `okta-react-native`, I am not really sure what's the difference between them, but it would be good to keep the test lib consistent cross repos.

Reply:
👍  It definitely makes sense to break up type test into smaller files.

`tsd` seemed like a better choice as it does not depend on [deprecated](https://palantir.github.io/tslint/) tslint package. There is no much difference otherwise.